lib/text_sentinel.rb

```diff
@@ -45,7 +45,7 @@ def valid?
     seems_meaningful? &&
     seems_pronounceable? &&
     seems_unpretentious? &&
-    seems_quiet?
+    (seems_quiet? || SiteSetting.allow_uppercase_posts)
```

Comment:
I would write `(SiteSetting.allow_uppercase_posts || seems_quiet?)` so that we don't even call the `seems_quiet?` method when we allow all uppercase posts ;)